lib/user.rb

```diff
@@ -0,0 +1,19 @@
+class User
+  attr_reader :name, :jokes, :jokes_id
+
+  def initialize(name) 
+    @name = name
+    @jokes = []
+    @jokes_id = []
+  end
+
+  def learn(joke)
+    @jokes_id << joke.id
+    @jokes << joke 
+  end
+
+  def tell(friend, joke)
+    friend.jokes << joke 
+  end
+
```

Comment:
You are missing a method-- `joke_by_id`... it's at the end of iteration 3.  I assume that you didn't see it, and not a huge deal, but make sure to read the interaction pattern carefully 🙂